src/lib/storages/RedisProxyCache.ts

```diff
@@ -62,31 +64,46 @@ export class RedisProxyCache implements IProxyCache {
 
     const uniqueProxyIds = [...new Set(proxyIds)];
     const ttl = ttlSec ?? this.defaultTtlSec;
-    const [addedCount] = await this.executePipeline([
-      ['sadd', key, uniqueProxyIds],
-      ['expire', key, ttl],
+    const expiryTime = Date.now() + ttl * 1000;
+    const [addedCount, expirySetResult] = await Promise.all([
+      this.redisClient.sadd(key, uniqueProxyIds),
+      this.redisClient.set(expiryKey, expiryTime),
```

Comment:
Why don't you use only one request to Redis to execute several commands (e.g., using _executePipeline_ methods)?

Reply:
Pipelines don't work for multi-key operations in cluster mode.